core/src/main/java/io/fabric8/maven/core/handler/EnvVarHandler.java

```diff
@@ -53,6 +53,12 @@
                 .endValueFrom()
                 .build());
 
+        ret.add(
+            new EnvVarBuilder()
+                    .withName("FABRIC8_GENERATED")
```

Comment:
I wonder whether we could use some other kind of meta-data for a container to indicate that it's auto generated? As this env var is visible within the container it might have side effects (although unlikely) but at least it pollutes the env namespace and increases the descriptors size. I understand that a container within a pod cannot carry any annotations/labels.

However, we need this information only during the (internal) build up of the model, so it sounds a bit over the top to make this information persistent. I suspect that the Kubernetes model objects don't carry any 'transient' meta information (which would be useful here), but maybe we could, in a postrpocess step, remove our markers before writing out the resource files ?

Reply:
Yes, it was my concern too. I think having a enricher that removes the env variable at the end of the build is ok. However, I think that I can also use a annotation at pod level to specify the name of the container that is generated by fabric8. Let's keep this on hold and I'll try to provide a different implementation. I don't like the added env variable either.